sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -63,84 +63,129 @@ static char x509_trust_pem_file[256] = { 0 };
 static az_iot_provisioning_client provisioning_client;
 static MQTTClient mqtt_client;
 
-static void sleep_seconds(uint32_t seconds)
-{
-#ifdef _WIN32
-  Sleep((DWORD)seconds * 1000);
-#else
-  sleep(seconds);
-#endif
-}
-
-// Read OS environment variables using stdlib function
+//
+// Configuration and connection functions
+//
+static az_result read_configuration_and_init_client();
 static az_result read_configuration_entry(
-    const char* name,
     const char* env_name,
     char* default_value,
     bool hide_value,
     az_span buffer,
-    az_span* out_value)
+    az_span* out_value);
+static int connect_device();
+static int subscribe();
+
+//
+// Messaging functions
+//
+static int register_device();
+static int get_operation_status();
+static void sleep_for_seconds(uint32_t seconds);
+static void print_az_span(az_span span);
+
+int main()
 {
-  printf("%s = ", name);
-  char* env = getenv(env_name);
+  int rc;
 
-  if (env != NULL)
+  // Read in the necessary environment variables and initialize the az_iot_provisioning_client
+  if (az_failed(rc = read_configuration_and_init_client()))
   {
-    printf("%s\n", hide_value ? "***" : env);
-    az_span env_span = az_span_from_str(env);
-    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(env_span));
-    az_span_copy(buffer, env_span);
-    *out_value = az_span_slice(buffer, 0, az_span_size(env_span));
+    printf(
+        "Failed to read configuration from environment variables, az_result return code %04x\n",
+        rc);
+    return rc;
   }
-  else if (default_value != NULL)
+
+  // Get the MQTT client id used for the MQTT connection
+  if (az_failed(
+          rc = az_iot_provisioning_client_get_client_id(
```

Comment:
Nit: formatting - would az_failed and the func call fit on a single line?

Reply:
clang format likes it on separate :/